File: app/src/main/java/com/whiteside/insta/MainActivity.java
package com.whiteside.insta;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;
import com.whiteside.insta.databinding.ActivityMainBinding;
import com.whiteside.insta.ui.profile.LoginActivity;


public class MainActivity extends AppCompatActivity {

    private FirebaseAuth auth;
    private ActivityMainBinding bind;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        bind = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(bind.getRoot());

        SecretData.initializeFirebase(this);

        auth = FirebaseAuth.getInstance();
    }

    @Override
    protected void onStart() {
        super.onStart();

        viewSplashPage();
    }

    private void viewSplashPage() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                checkLogin();
            }
        }, 2000);
    }

    private void checkLogin() {
        if (auth.getCurrentUser() != null) {
            goFeed();
        } else {
            goLogin();
        }
    }

    private void goFeed() {
        Intent intent = new Intent(this, WallActivity.class);
        startActivity(intent);
        finish();
    }

    private void goLogin() {
        Intent intent = new Intent(this, LoginActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/whiteside/insta/MainActivity.kt
package com.whiteside.insta

import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import com.airbnb.lottie.compose.LottieAnimation
import com.airbnb.lottie.compose.LottieCompositionSpec
import com.airbnb.lottie.compose.animateLottieCompositionAsState
import com.airbnb.lottie.compose.rememberLottieComposition
import com.google.firebase.auth.FirebaseUser
import com.whiteside.insta.db.network.abstract_.AuthManager
import com.whiteside.insta.ui.home.HomeActivity
import dagger.hilt.android.AndroidEntryPoint
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject


@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private val mainViewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            SplashScreen()
        }
    }

    @Composable
    fun SplashScreen() {
        val composition by rememberLottieComposition(LottieCompositionSpec.RawRes(R.raw.splash))
        val progress by animateLottieCompositionAsState(composition)
        val context = LocalContext.current

        Box {
            LottieAnimation(
                composition,
                progress,
            )

            Text(
                text = "Skip intro",
                modifier = Modifier
                    .padding(20.dp)
                    .clickable {
                        startActivity(Intent(context, HomeActivity::class.java))
                        finish()
                    }
            )

            if (progress >= 1f) {
                startActivity(Intent(context, HomeActivity::class.java))
                finish()
            }
        }
    }

    @HiltViewModel
    internal class MainViewModel @Inject constructor(
        val authManager: AuthManager,
    ) : ViewModel() {
        var user = mutableStateOf<FirebaseUser?>(null)

        fun getCurrentUser() {
            user.value = authManager.getCurrentUser()
        }

        fun signIn() {
            authManager.signInWithEmailAndRandomPassword(
                "hassan.shaban.harera@gmail.com"
            )
                .addOnSuccessListener {
                    user.value = it.user
                }
        }
    }
}